### ISSUE TITLE
 Can't call numpy() on Variable that requires grad. Use var.detach().numpy() instead.

### ISSUE THREAD

**Issue by skylook** [utiasSTARS/liegroups#4]
When I use liegroups in a loss function, it reports error as follows: 
Can't call numpy() on Variable that requires grad. Use var.detach().numpy() instead.

Here is the code:
```python
    print('rotation_ab_pred size = {} type = {}'.format(rotation_ab_pred.size(), rotation_ab_pred.type()))
    rvec_ab_pred = SO2.from_matrix(rotation_ab_pred)
```

And the result:

`rotation_ab_pred size = torch.Size([32, 2, 2]) type = torch.cuda.FloatTensor`

**Comment by leeclemnet** [comment]
There are a couple places where we convert to numpy to use np.linalg functions (specifically the determinant and SVD) to validate and normalize rotation matrices. I added the missing detach() calls in https://github.com/utiasSTARS/liegroups/commit/c2b479ec6179ebbdec77ab5c4054ac7e34758719#diff-eb304ee226fd0486879830ba65b8f46a so it should work as expected now.

As an alternative, you can also use the less safe constructor, which doesn't do any validity checking:
```python
rvec_ab_pred = SO2.from_matrix(rotation_ab_pred) # checks validity of matrix
rvec_ab_pred = SO2(rotation_ab_pred) # does not check validity of matrix
```
